spec/inputs/file_spec.rb

```diff
@@ -280,28 +301,104 @@
       FileUtils.rm_rf(tmpdir_path)
     end
 
-    let(:event_count) { 2 }
-
     it "collects separate multiple line events from each file" do
+      sleep 0.1
       writer_proc.call
+      sleep 0.3 # gotta wait for discovery to kick in and find the new files.
+      subject.stop
 
-      events = input(conf) do |pipeline, queue|
-        queue.size.times.collect { queue.pop }
-      end
+      expect(events.size).to eq(2)
 
-      expect(events.size).to eq(event_count)
+      e1, e2 = events
 
-      e1_message = events[0]["message"]
-      e2_message = events[1]["message"]
+      e1_message = e1["message"]
+      e2_message = e2["message"]
 
       # can't assume File A will be read first
       if e1_message.start_with?('line1.1-of-z')
+        expect(e1["path"]).to match(/z.log/)
+        expect(e2["path"]).to match(/a.log/)
         expect(e1_message).to eq("line1.1-of-z#{FILE_DELIMITER}  line1.2-of-z#{FILE_DELIMITER}  line1.3-of-z")
         expect(e2_message).to eq("line1.1-of-a#{FILE_DELIMITER}  line1.2-of-a#{FILE_DELIMITER}  line1.3-of-a")
       else
+        expect(e1["path"]).to match(/a.log/)
+        expect(e2["path"]).to match(/z.log/)
         expect(e1_message).to eq("line1.1-of-a#{FILE_DELIMITER}  line1.2-of-a#{FILE_DELIMITER}  line1.3-of-a")
         expect(e2_message).to eq("line1.1-of-z#{FILE_DELIMITER}  line1.2-of-z#{FILE_DELIMITER}  line1.3-of-z")
       end
     end
+
+    context "if auto_flush is enabled on the multiline codec" do
+      let(:writer_proc) do
+        -> do
+          File.open("#{tmpdir_path}/a.log", "a") do |fd|
+            fd.puts("line1.1-of-a")
+            fd.puts("  line1.2-of-a")
+            fd.puts("  line1.3-of-a")
+          end
+        end
+      end
+      let(:mlcodec) do
+        LogStash::Codecs::Multiline.new(
+          "pattern" => "^\s",
+          "what" => "previous",
+          "auto_flush_interval" => 2)
+      end
```

Comment:
I have seen the same codec defined in multiples place, maybe it could be move at the top level of the test?